src/state/viewer/types.ts

```diff
@@ -23,25 +25,20 @@ export interface ToggleAction {
 }
 
 export interface SetViewerStatusAction {
-    payload: ViewerStatusInfo;
+    payload: { status: ViewerStatus };
     type: string;
 }
 
-export interface ViewerStatusInfo {
-    htmlData?: string;
-    errorMessage?: string;
-    status: ViewerStatus;
-    onClose?: () => void;
+export interface SetErrorAction {
+    payload: ViewerError;
+    type: string;
 }
 
 export interface ViewerError {
-    htmlData?: string;
+    level: ErrorLevel;
```

Comment:
small thing, but if we're exporting FrontEndError, I think you could do `interface ViewerError extends FrontEndError`

Reply:
Ooh, I just tried it but I'm getting some errors because now I need to supply a `name` every time I use `setError`

![image](https://user-images.githubusercontent.com/12690133/141378250-7f718700-d007-4934-a501-6f8bdbd27605.png)